src/software/simulated_tests/simulated_test_fixture.cpp

```diff
@@ -179,7 +189,23 @@ void SimulatedTestFixture::runTest(
             continue;
         }
         validation_functions_done = tickTest(simulation_time_step, ai_time_step, world);
+
+        // Calculate tick durations
+        double duration_ms = ::TestUtil::millisecondsSince(start_tick_time);
+        max_tick_duration =
+            max_tick_duration > duration_ms ? max_tick_duration : duration_ms;
+        min_tick_duration =
+            min_tick_duration < duration_ms ? min_tick_duration : duration_ms;
+        tick_count++;
     }
+    // Output the tick duration results
+    double total_simulation_duration = ::TestUtil::millisecondsSince(start_time);
+    double avg_tick_duration         = total_simulation_duration / (double)tick_count;
```

Comment:
Can we sum up tick durations similar to how we calculate min/max to get a more accurate number? Seems like we wouldn't need to initialize tick count to 2 and have this note: `value is not accurate if stop_ai_on_start flag was used` (Also, use static_cast<double> instead of (double))

Reply:
Good idea, updated.